go/common/crypto/signature/signers/plugin/plugin.go

```diff
@@ -3,33 +3,68 @@ package plugin
 
 import (
 	"fmt"
-	"plugin"
+	"io"
+	"net/rpc"
+	"os/exec"
+	"sync"
+	"syscall"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
 
 	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
 )
 
 const (
-	// EntryPoint is the entry point (FactoryCtor) expected to be defined
-	// in each signer plugin.
-	EntryPoint = "GetPluginCtor"
-
-	// SignerName is the name used to identify the plugin signer.
+	// SignerName is the name used to identify the plugin backed signer.
 	SignerName = "plugin"
+
+	pluginName = "oasis_core_signer"
+)
+
+var (
+	_ signature.SignerFactory = (*wrapperFactory)(nil)
+	_ signature.Signer        = (*wrapperSigner)(nil)
+	_ plugin.Plugin           = (*signerPlugin)(nil)
+
+	// Sigh, yet another logging library. :(
+	nullLogger = hclog.NewNullLogger()
 )
 
-// FactoryCtor is the function signature of the EntryPoint symbol
-// that is expected to be defined in each plugin.
-type FactoryCtor func() signature.SignerFactoryCtor
+func handshakeConfigForName(name string) plugin.HandshakeConfig {
+	return plugin.HandshakeConfig{
+		ProtocolVersion:  1,
+		MagicCookieKey:   "OASIS_CORE_SIGNER_PLUGIN",
+		MagicCookieValue: name,
+	}
+}
 
 // FactoryConfig is the plugin factory configuration.
 type FactoryConfig struct {
+	// Name is the expected human readable name of the plugin (eg: "ledger", "memory")
```

Comment:
Why does the name need to be specified here? Shouldn't it be provided by the plugin?

Reply:
I could remove it entirely instead and just have every plugin named `signer` as far as the transport goes.  Figured I'd make users explicitly know what plugin the random binary they're passing to the node actually is.